src/bridging/BridgingSdk/getQuoteWithBridge.ts

```diff
@@ -197,20 +197,22 @@ export async function getQuoteWithBridge<T extends BridgeQuoteResult>(
     }
   }
 
+  const defaultGasLimit = bridgeHookSigner ? BigInt(hookEstimatedGasLimit) : undefined
+  log(`Using gas limit: ${defaultGasLimit}`)
   const result = await signHooksAndSetSwapResult(
     // Sign the hooks with bridgeHookSigner if provided
     bridgeHookSigner ? getSigner(bridgeHookSigner) : signer,
     // Use estimated hook gas limit if bridgeHookSigner is provided, so we don't have to estimate the hook gas limit twice
     // Moreover, since bridgeHookSigner is not the real signer, the estimation will fail
-    bridgeHookSigner ? BigInt(hookEstimatedGasLimit) : undefined,
+    defaultGasLimit,
   )
 
   return {
     swap: result.swapResult,
     bridge: result.bridgeResult,
     async postSwapOrderFromQuote(advancedSettings?: SwapAdvancedSettings) {
       // Sign the hooks with the real signer
-      const { swapResult } = await signHooksAndSetSwapResult(signer, undefined, advancedSettings)
+      const { swapResult } = await signHooksAndSetSwapResult(signer, defaultGasLimit, advancedSettings)
```

Comment:
It looks like the default gas limit was not being used unless `advancedSettings.quoteSigner` was passed. Even after passing, the second quote during `postSwapOrderFromQuote()` was also reverting and not using default gas limit since here it was `undefined`. Hence had to fix this

Came across this while testing CCTP which reverts in case the input amount deposit is zero

Reply:
But wouldn't the hook estimation on its own without the swap always fail, since the cowshed proxy won't have the tokens to be used for bridging? Particularly in CCTP's case since the input amount is replaced at runtime with balance, it fails at the CCTP contract. With Across, it wasn't failing I think because their contract don't have a non-zero check
Log from the test script with the reverting estimation tx: [cctp-amount-nonzero-revert.log](https://github.com/user-attachments/files/20477507/cctp-amount-nonzero-revert.log)
Log from cast simulation of the reverting estimation tx: [swapAndBridgeSdk-bungee-cctp.log](https://github.com/user-attachments/files/20477508/swapAndBridgeSdk-bungee-cctp.log)